lib/rules/block-attributes-order.js

```diff
@@ -0,0 +1,320 @@
+    order: [
+      WELL_KNOWN_TEMPLATE_ATTRS.functional,
+      WELL_KNOWN_TEMPLATE_ATTRS.lang,
+      WELL_KNOWN_TEMPLATE_ATTRS.src
+    ]
+  }
+])
+
+/**
+ *
+ * @param {string | string[]} item
+ * @return {RegExp | RegExp[]}
+ */
+function ordersToRegexOrders(item) {
+  return Array.isArray(item) ? item.map((x) => new RegExp(x)) : new RegExp(item)
+}
+
+/**
+ * @param {UserOrderOption} option
+ * @param {RuleContext} context
+ * @return {OrderOption}
+ */
+function normalizeUserOrderOption(option, context) {
+  return {
+    ...option,
+    selector: parseSelector(option.element, context),
+    regexOrder: option.order.map(ordersToRegexOrders)
+  }
+}
+
+/**
+ * @param {readonly UserOrderOption[] | undefined} options
+ * @param {RuleContext} context
+ * @returns {OrderOption[]}
+ */
+function normalizeUserOrderOptions(options, context) {
+  if (!options) {
+    return []
+  }
+  return options.map((option) => normalizeUserOrderOption(option, context))
+}
+
+/**
+ * @param {OrderOption} option
+ * @param {VElement} elementName
+ * @returns {boolean}
+ */
+function matchOrderOption(option, elementName) {
+  return option.selector.test(elementName)
+}
+
+/**
+ * @param {VAttribute | VDirective} attribute
+ * @param {OrderOption} orderOption
+ * @returns {number | undefined}
+ */
+function getMatchedAttributePosition(attribute, orderOption) {
+  if (attribute.directive) {
+    return undefined
+  }
+
+  const name = attribute.key.name
+  for (const [i, item] of orderOption.regexOrder.entries()) {
+    if (Array.isArray(item)) {
+      if (item.some((x) => x.test(name))) {
+        return i
+      }
+    } else if (item.test(name)) {
+      return i
+    }
+  }
+  return undefined
+}
+
+/**
+ * @param {VStartTag} node
+ * @param {OrderOption} orderOption
+ */
+function getAttributeAndPositionList(node, orderOption) {
+  /**
+   * @type {{ attr: (VAttribute | VDirective), position: number }[]}
+   */
+  const results = []
+  for (const attr of node.attributes) {
+    const position = getMatchedAttributePosition(attr, orderOption)
+    if (position == null) {
+      continue
+    }
+    results.push({ attr, position })
+  }
+  return results
+}
+
+/**
+ * @param {RuleContext} context - The rule context.
+ * @returns {RuleListener} AST event handlers.
+ */
+function create(context) {
+  const sourceCode = context.getSourceCode()
+  const normalizedDefaultOptions = normalizeUserOrderOptions(
+    defaultOptions,
+    context
+  )
+  const normalizedOrderOptions = normalizeUserOrderOptions(
+    context.options[0],
+    context
+  )
+
+  /**
+   * @param {VElement} element
+   * @returns {OrderOption | undefined}
+   */
+  function getMatchedOrderOption(element) {
+    const matchedOption = normalizedOrderOptions.find((option) =>
+      matchOrderOption(option, element)
+    )
+    if (matchedOption) {
+      return matchedOption
+    }
+    return normalizedDefaultOptions.find((option) =>
+      matchOrderOption(option, element)
+    )
```

Comment:
I don't understand why there is an `if` and the same function call again otherwise. Could it not be simplified to this?

```suggestion
    return normalizedDefaultOptions.find((option) =>
      matchOrderOption(option, element)
    )
```